tests/std/tests/GH_000940_missing_valarray_copy/test.cpp

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <iostream>
+#include <valarray>
+
+template <class T>
+bool eq(const std::valarray<T>& v, std::initializer_list<T> il) {
+    return std::equal(begin(v), end(v), il.begin(), il.end());
+}
+
+void test_slice() {
+    std::valarray<int> v{0, 1, 2, 3, 4};
+
+    std::slice_array<int> slice_array      = v[std::slice(2, 2, 2)];
+    std::slice_array<int> slice_array_copy = slice_array;
+    (void) slice_array_copy;
+
+    assert(eq(v, {0, 1, 2, 3, 4}));
+
+    std::slice_array<int> other_slice_array = v[std::slice(1, 2, 1)];
+    other_slice_array                       = slice_array;

```

Comment:
I think overlapping assignment *should* be UB. (the standard requires non-overlapping `valarray::operator=(const meow_array&)`, but apparently doesn't say anything about copy assignment of `moew_array`)

Reply:
agree